app/services/itemizable_update_service.rb

```diff
@@ -0,0 +1,49 @@
+module ItemizableUpdateService
+  # @param itemizable [Itemizable]
+  # @param type [Symbol] :increase or :decrease - if the original line items added quantities (purchases or
+  # donations), use :increase. If the original line_items reduced quantities (distributions) use :decrease.
+  # @param params [Hash] Parameters passed from the controller. Should include `line_item_attributes`.
+  def self.call(itemizable:, type: :increase, params: {})
+    StorageLocation.transaction do
+      item_ids = params[:line_items_attributes]&.values&.map { |i| i[:item_id].to_i } || []
+      Item.reactivate(item_ids)
+
+      from_location = to_location = itemizable.storage_location
+      to_location = StorageLocation.find(params[:storage_location_id]) if params[:storage_location_id]
+
+      increase_method = (type == :increase) ? :increase_inventory : :decrease_inventory
+      decrease_method = (type == :increase) ? :decrease_inventory : :increase_inventory
```

Comment:
Since the names flip around depending on the direction, how about we change to a name that describes the intent a bit better. Here's an idea:

```suggestion
      apply_change_method = (type == :increase) ? :increase_inventory : :decrease_inventory
      undo_change_method = (type == :increase) ? :decrease_inventory : :increase_inventory
```

Reply:
Love it!